acceptance-tests/apps/storageapp/internal/credentials/credentials.go

```diff
@@ -22,14 +23,46 @@ func Read() (StorageCredentials, error) {
 		return StorageCredentials{}, fmt.Errorf("error reading Storage service details")
 	}
 
+	switch svs[0].Label {
+	case "google-storage":
+		return readLegacyBrokerBinding(svs[0].Credentials)
+	default:
+		return readBinding(svs[0].Credentials)
+	}
+}
+
+func readBinding(creds any) (StorageCredentials, error) {
 	var r StorageCredentials
-	if err := mapstructure.Decode(svs[0].Credentials, &r); err != nil {
-		return StorageCredentials{}, fmt.Errorf("failed to decode credentials: %w", err)
+	if err := mapstructure.Decode(creds, &r); err != nil {
+		return r, fmt.Errorf("failed to decode credentials: %w", err)
 	}
 
 	if r.Credentials == "" || r.BucketName == "" {
-		return StorageCredentials{}, fmt.Errorf("parsed credentials are not valid")
+		return r, fmt.Errorf("parsed credentials are not valid")
+	}
+
+	return r, nil
+}
+
+func readLegacyBrokerBinding(creds any) (StorageCredentials, error) {
+	type legacyBindingData struct {
```

Comment:
non-blocking: since this type is only used once, it's possible to define it inline:
```go
var l struct {
  BucketName     string `mapstructure:"bucket_name"`
  PrivateKeyData string `mapstructure:"PrivateKeyData"`
}
```

Reply:
I like it. I'll change it. Thank you very much @blgm .